### DISCUSSION TITLE
Help in setting parameter allow_pickle for a function that returns numpy arrays

### DISCUSSION THREAD

**Discussion by sakthi-tech4th** [flyteorg/flyte#2904] [Q&A] [answered]
Hi,

I'm new to Flyte. I've created a task that returns two numpy arrays. However, I'm getting an error that says that the parameter allow_pickle has been set to False and should be set to True. However, that parameter is set when using numpy.load or numpy.save. My function uses neither. This is the output error message:
![image](https://user-images.githubusercontent.com/110414092/191395326-e934f8f4-fec1-4bcf-9bc3-ba5ca256a9c8.png)

My understanding is that Flyte task internally stores the return values and retrieves it. And it requires the parameter allow_pickle=True. However, I couldn't find any documentation on how to set this parameter. 

I greatly appreciate, any help in this regard.

Hope to hear soon from you,
Thanks

**Comment by samhita-alla** [comment, marked as answer]
@sakthi-tech4th, pickling isn't recommended by NumPy for security reasons: https://numpy.org/doc/stable/reference/generated/numpy.load.html. However, I can work on adding support for NumPy annotation with which you can set `allow_pickle`  to True. ``np.ndarray`` without an object dtype is a valid Flyte type, though.
